monkey/monkey_island/cc/agent_event_handlers/update_machine_hostname.py

```diff
@@ -0,0 +1,12 @@
+from common.agent_events import HostnameDiscoveryEvent
+from monkey_island.cc.repository import AgentMachineFacade
+
+
+class update_machine_hostname:
+    def __init__(self, agent_machine_facade: AgentMachineFacade):
+        self._agent_machine_facade = agent_machine_facade
+
+    def __call__(self, event: HostnameDiscoveryEvent):
+        machine = self._agent_machine_facade.get_agent_machine(event.source)
```

Comment:
We shouldn't presume that the hostname is of the local machine. Can we use `event.target` instead?

Reply:
If it is the local machine, `event.target` will be `None`. We could add a check but it seems unnecessary to me right now. We can modify it when (if) we end up in a situation where this event can be from a machine about a different machine.